src/jenkins/BuildManifest.groovy

```diff
@@ -0,0 +1,55 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package jenkins
+
+import groovy.transform.InheritConstructors
+
+@InheritConstructors
+class BuildManifest extends Manifest {
+    class Build implements Serializable {
+        String name
+        String version
+        String platform
+        String architecture
+
+        Build(Map data) {
+            this.name = data.name
+            this.version = data.version
+            this.platform = data.platform
+            this.architecture = data.architecture
+        }
+    }
+
+    Build build
+
+    BuildManifest(Map data) {
+        super(data)
+
+        this.build = new BuildManifest.Build(data.build)
+    }
+
+    public String getArtifactRoot(String buildNumber) {
+        return [
+            this.build.name.toLowerCase().replaceAll(' ', '-'),
```

Comment:
Are we changing the name here from `bundle-build/bundle-build-dashboards` to `opensearch/opensearch-dashboards`?

Reply:
In the build manifest ([example](https://github.com/opensearch-project/opensearch-build/blob/main/tests/tests_ci_workflow/data/opensearch-1.1.0-x64-build-manifest.yml#L4)) this is "OpenSearch" or "OpenSearch Dashboards", so it becomes 'opensearch' and 'opensearch-dashboards'.

Reply:
I think our URL used to use the jenkins job names in the path which are `bundle-build` and `bundle-build-dashboards`,  and they help us distinguish between `OpenSearch` and `OpenSearch Dashboards`. I am good with these names change because they work for the same purpose. Just make sure we are aware of this.

Reply:
Oh I see! I didn't realize we're using the job name for this purpose, this is my bad. Let's not change this - I added a change that uses `env.JOB_NAME` instead, please re-review and thanks for catching this!